frontend/packages/dev-console/src/components/project-access/ProjectAccessForm.tsx

```diff
@@ -0,0 +1,149 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { Formik } from 'formik';
+import { Link } from 'react-router-dom';
+import { LoadingBox, PageHeading, ExternalLink } from '@console/internal/components/utils';
+import { getActiveNamespace } from '@console/internal/actions/ui';
+import { RoleBindingModel, RoleModel } from '@console/internal/models';
+import {
+  validationSchema,
+  UserRoleBinding,
+  filterRoleBindings,
+  getRolesWithNameChange,
+  getNewRoles,
+  getRemovedRoles,
+  getFinalRoles,
+  Verb,
+  Roles,
+  getUserRoleBindings,
+  sendRoleBindingRequest,
+} from './project-access-form-utils';
+import ProjectAccess from './ProjectAccess';
+
+export interface ProjectAccessFormProps {
+  formName: string;
+  namespace: string;
+  roleBindings?: { data: []; loaded: boolean; loadError: {} };
+}
+
+const ProjectAccessForm: React.FC<ProjectAccessFormProps> = ({
+  formName,
+  namespace,
+  roleBindings,
+}) => {
+  if (!roleBindings.loaded && _.isEmpty(roleBindings.loadError)) {
+    return <LoadingBox />;
+  }
+
+  const noProjectsAvailable = !_.isEmpty(roleBindings.loadError);
+
+  const filteredRoleBindings = filterRoleBindings(roleBindings, Roles);
+
+  const userRoleBindings: UserRoleBinding[] = getUserRoleBindings(filteredRoleBindings);
+
+  const initialValues = {
+    projectAccess: roleBindings.loaded && userRoleBindings,
+  };
+
+  const handleSubmit = (values, actions) => {
+    let newRoles = getNewRoles(initialValues.projectAccess, values.projectAccess);
+    let removeRoles = getRemovedRoles(initialValues.projectAccess, values.projectAccess);
+    if (_.isEmpty(newRoles) && _.isEmpty(removeRoles)) {
+      actions.setSubmitting(false);
+      actions.resetForm({
+        values: {
+          projectAccess: initialValues.projectAccess,
+        },
+        status: { success: `Successfully updated the ${formName}.` },
+      });
+      return;
+    }
+
+    const updateRoles = getRolesWithNameChange(newRoles, removeRoles);
+
+    if (!_.isEmpty(updateRoles)) {
+      newRoles = _.filter(
+        newRoles,
+        (o1) => !updateRoles.find((o2) => o1.roleBindingName === o2.roleBindingName),
+      );
+      removeRoles = _.filter(
+        removeRoles,
+        (o1) => !updateRoles.find((o2) => o1.roleBindingName === o2.roleBindingName),
+      );
+    }
+
+    const roleBinding = {
```

Comment:
nit: this object can be moved out of this function as a constant at the top of the file or into a constant file or types file.